source/slang/slang-check-conversion.cpp

```diff
@@ -201,11 +201,137 @@ namespace Slang
+    ConstructorDecl* SemanticsVisitor::_getSynthesizedConstructor(StructDecl* structDecl)
+    {
+        for (auto ctor : getMembersOfType<ConstructorDecl>(getASTBuilder(), structDecl, MemberFilterStyle::All))
+        {
+            if (ctor.getDecl()->findModifier<SynthesizedModifier>() != nullptr)
+            {
+                return ctor.getDecl();
+            }
+        }
+
+        return nullptr;
+    }
+
+    Expr* SemanticsVisitor::_prepareCtorInvokeExpr(Type* toType, const SourceLoc& loc, const List<Expr*>& coercedArgs)
+    {
+        auto* varExpr = getASTBuilder()->create<VarExpr>();
+        varExpr->type = (QualType)getASTBuilder()->getTypeType(toType);
+        varExpr->declRef = isDeclRefTypeOf<Decl>(toType);
+
+        auto* constructorExpr = getASTBuilder()->create<InvokeExpr>();
+        constructorExpr->functionExpr = varExpr;
+        constructorExpr->arguments.addRange(coercedArgs);
+        constructorExpr->loc = loc;
+        auto resolvedConstructorExpr = CheckExpr(constructorExpr);
+
+        return resolvedConstructorExpr;
+    }
+
+    // translation from initializer list to constructor invocation if the struct has constructor.
+    bool SemanticsVisitor::_invokeExprForExplicitCtor(Type* toType, InitializerListExpr* fromInitializerListExpr, Expr** outExpr)
+    {
+        if(auto toDeclRefType = as<DeclRefType>(toType))
+        {
+            auto toTypeDeclRef = toDeclRefType->getDeclRef();
+            if(auto toStructDeclRef = toTypeDeclRef.as<StructDecl>())
+            {
+                if (isFromStdLib(toStructDeclRef.getDecl()))
+                {
+                    // If the struct is from stdlib, we will not try to create a constructor.
+                    return false;
+                }
+
+                if (_hasExplicitConstructor(toStructDeclRef.getDecl()))
+                {
+                    auto ctorInvokeExpr = _prepareCtorInvokeExpr(toType, fromInitializerListExpr->loc, fromInitializerListExpr->args);
+                    if (outExpr && ctorInvokeExpr)
+                    {
+                        *outExpr = ctorInvokeExpr;
+                        return true;
+                    }
+                }
+            }
+        }
+        return false;
+    }
+
+    bool SemanticsVisitor::_invokeExprForSynthesizedCtor(
+        Type*                   toType,
+        InitializerListExpr*    fromInitializerListExpr,
+        Expr**                  outExpr)
+    {
+        StructDecl* structDecl = nullptr;
+        if(auto toDeclRefType = as<DeclRefType>(toType))
+        {
+            auto toTypeDeclRef = toDeclRefType->getDeclRef();
+            if(auto toStructDeclRef = toTypeDeclRef.as<StructDecl>())
+            {
+                structDecl = toStructDeclRef.getDecl();
+            }
+        }
+
+        if (!structDecl || isFromStdLib(structDecl))
+            return true;
+
+        auto synthesizedConstructor = _getSynthesizedConstructor(structDecl);
```

Comment:
This is not used?
Perhaps we need a version of `_prepareCtorInvokeExpr` where the ctor decl is known so we don't redundantly resolve overloads in `ResolveInvoke`.